### ISSUE TITLE
Memory leak with the Geodetic() transform

### ISSUE THREAD

**Issue by matvidal** [SciTools/cartopy#1557]
### Description

When I plot polygons using the Geodetic transformation, the memory usage increases steadily over time. If I remove the transformation, there is no memory leakage but the polygons don't close correctly (for example instead of having 2 semi-spheres on both borders of the map, I get a "rectangle" with concave borders on both sides).

For the next example, I created 2 polygons so that the memory leakage is more evident. (In my program I can have multiple polygons representing the coverage of multiple satellites).

#### Code to reproduce

```Python
from cartopy.crs import Geodetic, PlateCarree
from cartopy.geodesic import Geodesic
from matplotlib.patches import Polygon
from matplotlib.pyplot import figure, ion, show, pause
from numpy import asarray
from random import randrange

fig = figure(figsize=(16, 8))
ax = fig.add_axes([0, 0, 1, 1], projection=PlateCarree(), frameon=False)
ax.stock_img()

init_xy = asarray(Geodesic().circle(lon=0, lat=0, radius=2500000,
                                    n_samples=200, endpoint=True))
ax_cov = []
ax_cov.append(ax.add_patch(Polygon(init_xy, color='w', alpha=0.2,
                                   transform=Geodetic()))) # <-- These transformations
ax_cov.append(ax.add_patch(Polygon(init_xy, color='k', alpha=0.2,
                                   transform=Geodetic()))) # <-- These transformations

def randomData():
    lon = randrange(-180, 180)
    lat = randrange(-90, 90)
    radius = randrange(2300000, 2600000)
    return lon, lat, radius

def coverage(n):
    lon, lat, radius = randomData()
    xy_data = asarray(Geodesic().circle(lon=lon, lat=lat, radius=radius,
                                        n_samples=200, endpoint=True))
    ax_cov[n].set_xy(xy_data)

ion()
show()

for i in range(0, 500):
    coverage(0)
    coverage(1)
    fig.canvas.draw_idle()
    pause(0.001)
```

#### Part of the code to change to remove the leakage (but with an incorrect fill)

```Python
ax_cov.append(ax.add_patch(Polygon(init_xy, color='w', alpha=0.2)))
ax_cov.append(ax.add_patch(Polygon(init_xy, color='k', alpha=0.2)))
```

#### Traceback 

```

```

<details>
 <summary>Full environment definition</summary>

### Operating system
Ubuntu 19.10

### Cartopy version
cartopy 0.17.0

### Python version
Python 3.7.5

### conda list

```
```

### pip list

```
```

</details>

**Comment by greglucas** [comment]
I couldn't see any noticeable leak on my system with that example, but my guess is that you're just building up a large cache of transformed geometries in the `_PATH_TRANSFORM_CACHE`  variable: https://github.com/SciTools/cartopy/blob/466c901f8f38ab4a26fd2736ccf49c98086de03b/lib/cartopy/mpl/geoaxes.py#L48

Explained here why there is caching
https://github.com/SciTools/cartopy/blob/466c901f8f38ab4a26fd2736ccf49c98086de03b/lib/cartopy/mpl/geoaxes.py#L143-L148
But, I don't see any mechanism to disable that caching easily.

**Comment by matvidal** [comment]
On my system, the example increases my RAM usage in around 3 MB. I know, it doesn't sound that bad but imagine I run a similar program with more than 2 polygons for hours. That is my concern. And when I remove the "transform=Geodetic()" argument, my memory doesn't change even in 0.x MB. 

I am curious, @greglucas , that in your system you don't see any noticeable leak. That makes me think that maybe the problem is not in my program, but in the version of the libraries that I am using. For example, when I install proj-bin, I get the 5.2.0-1 version. Can that be the reason?

**Comment by matvidal** [comment]
I just found the problem. I upgraded my **Cartopy** library from **0.17.0** to **0.18.0**, and the memory leak disappeared. I even increased the iterations of the for loop (where I updated the polygons) of the example case from 500 to 10000, and the memory remained at 100.9 MB.

I want to thank you @greglucas for taking your time.
In case anyone else bumps into the same issue, first check your Cartopy version:
```Python
import cartopy
cartopy.__version__
```
